[codex] Fix Feature Flags documentation guidance (#36573)
DataDog/documentation: content/en/feature_flags/server/ruby.md

## Overview

> This page describes how to instrument your Ruby application with the Datadog Feature Flags SDK. The Ruby SDK integrates with [OpenFeature][3], an open standard for feature flag management, and receives flag updates through Remote Configuration in the Datadog Ruby tracer (`datadog` gem).

Author: 🤖 **Context from Codex:** Clarifies that the Ruby page is talking about the Datadog Ruby tracer (`datadog` gem), not a generic Datadog SDK. Source: `DataDog/dd-trace-rb@master:lib/datadog/open_feature/provider.rb:8-10`.

## Prerequisites

Before setting up the Ruby Feature Flags SDK, ensure you have:

- **Datadog Agent** version 7.55 or later with [Remote Configuration][1] enabled
- **Datadog [API key][4]** configured on the Agent
- **Datadog Ruby SDK** `datadog` version 2.24.0 or later
- **OpenFeature Ruby SDK** `openfeature-sdk` version 0.5.1 or later if you need flag evaluation metrics support
- **Service and environment configured** - Feature flags are targeted by service and environment

> - **Supported operating system** - Production support is currently limited to [Linux operating systems][2]. macOS can be used for local development when a compatible prebuilt native artifact is available. Windows and macOS are not natively supported production targets, but Dockerized Linux environments running on those operating systems are.

Author: 🤖 **Context from Codex:** Keeps Linux as the production support boundary while documenting the macOS local-development nuance you clarified. Public source shows the OpenFeature component depends on MRI and the loaded `libdatadog` native layer: `DataDog/dd-trace-rb@master:lib/datadog/open_feature/component.rb:28-38`.

> - **Supported operating system** - Production support is currently limited to [Linux operating systems][2]. macOS can be used for local development when a compatible prebuilt native artifact is available. Windows and macOS are not natively supported production targets, but Dockerized Linux environments running on those operating systems are.

Reviewer: Two issues: "currently" should be removed, and the bullet covers three different facts (Linux-only prod, macOS dev artifact, Docker workaround) that don't sit cleanly together. Restructure for clearer scope:

```suggestion
- **Supported operating system** - Production support is limited to [Linux operating systems][2]. macOS and Windows are not natively supported production targets, but Dockerized Linux environments running on those operating systems are. For local development on macOS, you can use a compatible prebuilt native artifact when one is available.
```

Feature Flagging is provided by Application Performance Monitoring (APM). To integrate APM into your application with feature flagging support, install the required gems and configure Remote Configuration with OpenFeature support.

> ```shell

Author: 🤖 **Context from Codex:** Ruby tracing v2 uses the `datadog` gem, while `ddtrace` is the legacy v1 gem. The surrounding code already requires `datadog`, so the install command should match.

### Remote Configuration connection issues

> Check the Datadog Ruby tracer logs for Remote Configuration status:

Author: 🤖 **Context from Codex:** Uses `Datadog Ruby tracer logs` here so troubleshooting points at the component that owns Remote Configuration status in Ruby.
